### PR TITLE
Node agnostic client

### LINKED ISSUE #4336: hive client usage reporting doesn't work with cloudflare workers
Hello, while trying to use hive usage reporting with coudflare, I get the following error : 

```
 (error) [hive][info] Error There is no suitable adapter to dispatch the request since :
- adapter xhr is not supported by the environment
- adapter http is not available in the build
✘ [ERROR]   ReferenceError: global is not defined
```

This is what the handler looks like : 

```ts
const handler = {
  async fetch(request: Request, env, context) {
    const cloudflare = { env, context };
    const yoga =   createYoga({
    schema,
    plugins: [
      useHive({
        token: env.HIVE_TOKEN,
        usage: {
          clientInfo() {
            return {
              name: `${env.ENVIRONMENT}-api`,
              version: '0.0.1'
            };
          }
        }
      })
    ],
    logging: true
  });
    const response = await yoga(request, [{ request, cloudflare }]);
    return response;
  }
} satisfies ExportedHandler<Env>
```

From what I can see it's because the plugin use Axios instead of fetch. 
Is there an available workaround ? Or would that requires a fetch based rewrite ?

### PR DESCRIPTION
**Pull request by kamilkisiela**
- [x] document Node 16+ requirement
- [x] document Yoga.dispose()
- [x] document new `printTokenInfo` option
- [x] document that token info is only available in debug mode or when printTokenInfo is enabled

Fixes #4336 #2492

### CODE REVIEW

File: packages/libraries/client/src/internal/http-client.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    timeout?: number;
+  },
+) {
+  const controller = new AbortController();
+  let timeoutId: ReturnType<typeof setTimeout> | undefined = undefined;
+  const responsePromise = fetch(endpoint, {
+    method: config.method,
+    body: config.body,
+    headers: config.headers,
+    signal: controller.signal,
```
Comment (ardatan) [inline packages/libraries/client/src/internal/http-client.ts:48]: You can use `AbortSignal.timeout` instead;
https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/timeout_static

File: packages/libraries/client/src/internal/utils.ts [hunk 2]
```diff
@@ -1,7 +1,48 @@
...
 import { hiveClientSymbol } from '../client.js';
 import type { HiveClient, HivePluginOptions } from './types.js';
 
+export const isCloudflareWorker = Boolean(
+  typeof caches !== 'undefined' && 'default' in caches && caches.default,
+);
+
+async function digest(algo: 'SHA-256' | 'SHA-1', output: 'hex' | 'base64', data: string) {
+  if (typeof crypto !== 'undefined' && typeof TextEncoder !== 'undefined') {
+    const buffer = await crypto.subtle.digest(algo, new TextEncoder().encode(data));
```
Comment (ardatan) [inline packages/libraries/client/src/internal/utils.ts:10]: We have this in `@whatwg-node/fetch`

Comment (kamilkisiela) [reply packages/libraries/client/src/internal/utils.ts:10]: super weird to use WebCrypto API from `whatwg-node/fetch` :))

File: packages/libraries/client/src/internal/http-client.ts (hunk 1 above)
Comment (kamilkisiela) [reply packages/libraries/client/src/internal/http-client.ts:48]: It was added in v16.14.0 so I cannot really use it if I want to support Node v16

File: packages/libraries/client/src/internal/utils.ts (hunk 2 above)
Comment (ardatan) [reply packages/libraries/client/src/internal/utils.ts:10]: But we already have a dependency here :)
Also some bundlers can complain about the dynamic import below;
https://github.com/kamilkisiela/graphql-hive/pull/4494/files/840154d858b4708d6efb9d7bb1c67ae0ecf6eaa9#diff-1920db0a7d3df2907cd2cc1ed61b516ded3145e21a7487384f60b5d17854d10fR18

File: packages/libraries/client/README.md [hunk 3]
```diff
@@ -48,6 +48,35 @@ const server = createYoga({
...
+
+If you're using Cloudflare Workers, you can use the following code:
+
+```typescript
+import { useYogaHive, createYogaHive, useYogaHive } from '@graphql-hive/client'
+import { createYoga } from 'graphql-yoga'
+
+const hive = createYogaHive({
+  enabled: true, // Enable/Disable Hive Client
+  token: 'YOUR-TOKEN',
```
Comment (Hebilicious) [inline packages/libraries/client/README.md:61]: For cloudflare workers this needs to happen within the fetch handler as the env variables are not available globally. Could you consider an alternative API to pass the token down ? Maybe try to read it automatically from the cloudflare env assuming a specific variable name ?

Comment (kamilkisiela) [reply packages/libraries/client/README.md:61]: You can run `createYogaHive` within fetch, it has no overhead really

**Pull request merged**